src/plumbing/transcoding.c

```diff
@@ -1010,8 +1010,15 @@ transcoder_stream_video(transcoder_t *t, transcoder_stream_t *ts, th_pkt_t *pkt)
       octx->qmin           = 1;
       octx->qmax           = FF_LAMBDA_MAX;
 
-      octx->bit_rate       = 2 * octx->width * octx->height;
-      octx->rc_max_rate    = 4 * octx->bit_rate;
+      if (t->t_props.tp_vbitrate == 0) {
+        octx->bit_rate       = 2 * octx->width * octx->height;
+        octx->rc_max_rate    = 4 * octx->bit_rate;
+
+      } else {
+        octx->bit_rate       = t->t_props.tp_vbitrate * 1000;
+        octx->rc_max_rate    = octx->bit_rate * 1.05;
```

Comment:
Might be worth increasing this factor a little bit since most codecs perform much better if you give them a little headroom. If you're concerned that you'll use too much bandwidth, make the entered value the max rate and do the inverse of what you're doing here for `octx->bit_rate`.